gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/GobblinMCEWriter.java

```diff
@@ -344,12 +347,13 @@ private void flush(String dbName, String tableName) throws IOException {
           writer.flush(dbName, tableName);
         } catch (IOException e) {
           meetException = true;
-          addException(e, tableString, dbName, tableName);
+          writer.reset(dbName, tableName);
+          addOrThrowException(e, tableString, dbName, tableName);
         }
       }
     }
-    if (!meetException && datasetErrorMap.containsKey(tableOperationTypeMap.get(tableString).datasetName)
-        && datasetErrorMap.get(tableOperationTypeMap.get(tableString).datasetName).containsKey(tableString)) {
+    if (!meetException && datasetErrorMap.containsKey(tableOperationTypeMap.get(tableString).datasetPath)
```

Comment:
So the if condition is hard to understand. What is the scenario this is trying to address? Can we add a comment here to help the reader understand?